tests/applications/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+##############################################################################
+# Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+# project contributors. See the COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (MIT)
+##############################################################################
+set (test_applications_depends umpire)
+
+blt_add_executable(
+  NAME pool_allocator_stress
+  SOURCES pool_allocator_stress.cpp
+  DEPENDS_ON ${test_applications_depends})
+list(APPEND umpire_test_applications pool_allocator_stress)
+
+if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+  blt_add_target_link_flags(
+    TO pool_allocator_stress
+    FLAGS "-rdynamic -ldl")
```

Comment:
We can use `CMAKE_DL_LIBS` instead of `-ldl`: https://cmake.org/cmake/help/latest/variable/CMAKE_DL_LIBS.html

Reply:
Done!

Reply:
Wait.  This isn't right.  Since there can be multiple library names, I need to find a different way to do this.  I'm not sure how to do this with `blt_add_target_link_flags`.  I must need to use something different.

Reply:
Okay @davidbeckingsale, please go ahead and review.  I think this should be ready.